From dd5b1bcf4b16d723826bbeefda1d59ba980b60dc Mon Sep 17 00:00:00 2001
From: Larry Finger <Larry.Finger@lwfinger.net>
Date: Sat, 15 Mar 2014 22:00:20 -0500
Subject: [PATCH 331/390] Remove CONFIG_WOWLAN

This is always the same as CONFIG_WAKE_ON_WLAN.

Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>
---
 core/rtw_mlme.c         |  4 ++--
 hal/rtl8192d_cmd.c      |  4 ++--
 hal/rtl8192d_hal_init.c | 16 ++++++++--------
 hal/usb_halinit.c       | 28 ++++++++++++++--------------
 include/autoconf.h      |  7 +------
 include/hal_intf.h      |  4 ++--
 include/rtl8192d_cmd.h  | 12 ++++++------
 include/rtl8192d_hal.h  |  4 ++--
 include/rtw_pwrctrl.h   |  4 ++--
 os_dep/ioctl_linux.c    | 12 ++++++------
 os_dep/usb_intf.c       | 20 ++++++++++----------
 11 files changed, 55 insertions(+), 60 deletions(-)

--- a/drivers/staging/rtl8192du/core/rtw_mlme.c
+++ b/drivers/staging/rtl8192du/core/rtw_mlme.c
@@ -1179,9 +1179,9 @@ void rtw_indicate_disconnect(struct rtw_
 		rtw_led_control(padapter, LED_CTL_NO_LINK);
 		rtw_clear_scan_deny(padapter);
 	}
-#ifdef CONFIG_WOWLAN
+#ifdef CONFIG_WAKE_ON_WLAN
 	if (padapter->pwrctrlpriv.wowlan_mode == false)
-#endif /* CONFIG_WOWLAN */
+#endif /* CONFIG_WAKE_ON_WLAN */
 		rtw_lps_ctrl_wk_cmd(padapter, LPS_CTRL_DISCONNECT, 1);
 }
 
--- a/drivers/staging/rtl8192du/hal/rtl8192d_cmd.c
+++ b/drivers/staging/rtl8192du/hal/rtl8192d_cmd.c
@@ -859,7 +859,7 @@ int reset_tsf(struct rtw_adapter * adapt
 
 #endif	/*  CONFIG_CONCURRENT_MODE */
 
-#ifdef CONFIG_WOWLAN
+#ifdef CONFIG_WAKE_ON_WLAN
 
 void rtl8192d_set_wowlan_cmd(struct rtw_adapter* padapter)
 {
@@ -950,4 +950,4 @@ void rtl8192d_set_wowlan_cmd(struct rtw_
 	return ;
 }
 
-#endif  /* CONFIG_WOWLAN */
+#endif  /* CONFIG_WAKE_ON_WLAN */
--- a/drivers/staging/rtl8192du/hal/rtl8192d_hal_init.c
+++ b/drivers/staging/rtl8192du/hal/rtl8192d_hal_init.c
@@ -275,11 +275,11 @@ static bool get_fw_from_file(struct rtw_
 	struct device *device = dvobj_to_dev(dvobj);
 	const struct firmware *fw;
 	bool rtstatus = false;
-#ifdef CONFIG_WOWLAN
+#ifdef CONFIG_WAKE_ON_WLAN
 	const char *fw_name = "rtlwifi/rtl8192dufw_wol.bin";
 #else
 	const char *fw_name = "rtlwifi/rtl8192dufw.bin";
-#endif /* CONFIG_WOWLAN */
+#endif /* CONFIG_WAKE_ON_WLAN */
 
 	if (request_firmware(&fw, fw_name, device) || !fw) {
 		pr_err("Firmware %s not available\n", fw_name);
@@ -319,10 +319,10 @@ int FirmwareDownload92D(struct rtw_adapt
 	struct hal_data_8192du *pHalData = GET_HAL_DATA(adapter);
 	u8 *FwImage;
 	u32 FwImageLen;
-#ifdef CONFIG_WOWLAN
+#ifdef CONFIG_WAKE_ON_WLAN
 	u8 *FwImageWoWLAN;
 	u32 FwImageWoWLANLen;
-#endif /* CONFIG_WOWLAN */
+#endif /* CONFIG_WAKE_ON_WLAN */
 	struct rt_8192d_firmware_hdr *pFwHdr = NULL;
 	u8 *pFirmwareBuf;
 	u32 FirmwareLen;
@@ -354,7 +354,7 @@ int FirmwareDownload92D(struct rtw_adapt
 		FirmwareLen = FirmwareLen -32;
 	}
 
-#ifdef CONFIG_WOWLAN
+#ifdef CONFIG_WAKE_ON_WLAN
 	/* write 0x5 BIT(3), don't suspend to reset MAC */
 	if (bUsedWoWLANFw) {
 		u8 test;
@@ -363,7 +363,7 @@ int FirmwareDownload92D(struct rtw_adapt
 		rtw_write8(adapter, REG_APS_FSMCO+1, test);
 	}
 
-#endif /* CONFIG_WOWLAN */
+#endif /* CONFIG_WAKE_ON_WLAN */
 	ACQUIRE_GLOBAL_MUTEX(GlobalMutexForFwDownload);
 	if (pHalData->MacPhyMode92D == DUALMAC_DUALPHY ||
 	    pHalData->MacPhyMode92D == DUALMAC_SINGLEPHY) {
@@ -457,7 +457,7 @@ Exit:
 	return rtStatus;
 }
 
-#ifdef CONFIG_WOWLAN
+#ifdef CONFIG_WAKE_ON_WLAN
 void InitializeFirmwareVars92D(struct rtw_adapter *adapter)
 {
 	struct hal_data_8192du *pHalData = GET_HAL_DATA(adapter);
@@ -499,7 +499,7 @@ void SetFwRelatedForWoWLAN8192DU(struct
 		InitializeFirmwareVars92D(padapter);
 	}
 }
-#endif /* CONFIG_WOWLAN */
+#endif /* CONFIG_WAKE_ON_WLAN */
 
 /* chnl" begins from 0. It's not a real channel. */
 /* channel_info[chnl]" is a real channel. */
--- a/drivers/staging/rtl8192du/hal/usb_halinit.c
+++ b/drivers/staging/rtl8192du/hal/usb_halinit.c
@@ -408,7 +408,7 @@ static u16 calc_crc(u8 *pdata, int lengt
 	return CRC;
 }
 
-#ifdef CONFIG_WOWLAN
+#ifdef CONFIG_WAKE_ON_WLAN
 static int rtw_wowlan_set_pattern(struct rtw_adapter *padapter , u8 *pbuf) {
 	struct pwrctrl_priv *pwrpriv =&padapter->pwrctrlpriv;
 	int res = 0, crc_idx;
@@ -567,7 +567,7 @@ void rtw_wowlan_reload_pattern(struct rt
 
 	}
 }
-#endif /* CONFIG_WOWLAN */
+#endif /* CONFIG_WAKE_ON_WLAN */
 
 /*  */
 /*  */
@@ -1398,7 +1398,7 @@ static void init_hwled(struct rtw_adapte
 			return;
 }
 
-#ifdef CONFIG_WOWLAN
+#ifdef CONFIG_WAKE_ON_WLAN
 static void dump_wakup_reason(struct rtw_adapter *padapter)
 {
 	switch (rtw_read8(padapter, REG_WOWLAN_REASON))
@@ -1429,7 +1429,7 @@ static void dump_wakup_reason(struct rtw
 			break;
 	}
 }
-#endif /* CONFIG_WOWLAN */
+#endif /* CONFIG_WAKE_ON_WLAN */
 
 static u32 rtl8192du_hal_init(struct rtw_adapter *padapter)
 {
@@ -1484,7 +1484,7 @@ static u32 rtl8192du_hal_init(struct rtw
 	val8 &= (~BIT7);
 	rtw_write8(padapter, 0x0003, val8);
 
-#ifdef CONFIG_WOWLAN
+#ifdef CONFIG_WAKE_ON_WLAN
 	if (rtw_read8(padapter, REG_MCUFWDL)&BIT7)
 	{
 		u8 reg_val = 0;
@@ -1508,7 +1508,7 @@ static u32 rtl8192du_hal_init(struct rtw
 		reg_val |= BIT(4) | BIT(7);
 		rtw_write8(padapter, REG_SYS_FUNC_EN+1, reg_val);
 	}
-#endif /* CONFIG_WOWLAN */
+#endif /* CONFIG_WAKE_ON_WLAN */
 	/* mac status: */
 	/* 0x81[4]:0 mac0 off, 1:mac0 on */
 	/* 0x82[4]:0 mac1 off, 1: mac1 on. */
@@ -1841,7 +1841,7 @@ exit:
 
 	DBG_8192D("%s in %dms\n", __func__, rtw_get_passing_time_ms(init_start_time));
 
-#ifdef CONFIG_WOWLAN
+#ifdef CONFIG_WAKE_ON_WLAN
 	if (padapter->pwrctrlpriv.wowlan_mode == true)
 		dump_wakup_reason(padapter);
 	{
@@ -1854,7 +1854,7 @@ exit:
 	/* prevent 8051 to be reset by PERST# wake on wlan by Alex & Baron */
 	/* rtw_write8(padapter, REG_RSV_CTRL, 0x20); */
 	/* rtw_write8(padapter, REG_RSV_CTRL, 0x60); */
-#endif /*  CONFIG_WOWLAN */
+#endif /*  CONFIG_WAKE_ON_WLAN */
 	return status;
 }
 
@@ -2708,7 +2708,7 @@ _ReadLEDSetting(
 	}
 }
 
-#ifdef CONFIG_WOWLAN
+#ifdef CONFIG_WAKE_ON_WLAN
 static void
 _ReadWOWLAN(
 	struct rtw_adapter *	adapter,
@@ -2726,7 +2726,7 @@ _ReadWOWLAN(
 		DBG_8192D("efuse remote wakeup =%d\n", adapter->pwrctrlpriv.bSupportRemoteWakeup);
 	}
 }
-#endif /* CONFIG_WOWLAN */
+#endif /* CONFIG_WAKE_ON_WLAN */
 
 static void _InitadapterVariablesByPROM(
 	struct rtw_adapter *	adapter,
@@ -2742,9 +2742,9 @@ static void _InitadapterVariablesByPROM(
 	rtl8192d_EfuseParseChnlPlan(adapter, PROMContent, AutoloadFail);
 	_ReadBoardType(adapter, PROMContent, AutoloadFail);
 	_ReadLEDSetting(adapter, PROMContent, AutoloadFail);
-#ifdef CONFIG_WOWLAN
+#ifdef CONFIG_WAKE_ON_WLAN
 	_ReadWOWLAN(adapter, PROMContent, AutoloadFail);
-#endif /* CONFIG_WOWLAN */
+#endif /* CONFIG_WAKE_ON_WLAN */
 }
 
 static void _ReadPROMContent(
@@ -4229,7 +4229,7 @@ static void SetHwReg8192DU(struct rtw_ad
 			}
 			break;
 		case HW_VAR_WOWLAN:
-#ifdef CONFIG_WOWLAN
+#ifdef CONFIG_WAKE_ON_WLAN
 			{
 				struct wowlan_ioctl_param *poidparam;
 
@@ -4332,7 +4332,7 @@ static void SetHwReg8192DU(struct rtw_ad
 					adapter->pwrctrlpriv.wowlan_mode = false;
 			}
 			break;
-#endif /* CONFIG_WOWLAN */
+#endif /* CONFIG_WAKE_ON_WLAN */
 		case HW_VAR_CHECK_TXBUF:
 #if defined(CONFIG_CONCURRENT_MODE) || defined(CONFIG_DUALMAC_CONCURRENT)
 			{
--- a/drivers/staging/rtl8192du/include/autoconf.h
+++ b/drivers/staging/rtl8192du/include/autoconf.h
@@ -23,15 +23,10 @@
 
 #define CONFIG_IOCTL_CFG80211 1
 
-/*
- * Internal  General Config
- */
-#define CONFIG_PWRCTRL	1
-
 //#define CONFIG_WAKE_ON_WLAN
 
 #ifdef CONFIG_WAKE_ON_WLAN
-#define CONFIG_WOWLAN 1
+#define CONFIG_WAKE_ON_WLAN 1
 #endif /* CONFIG_WAKE_ON_WLAN */
 
 #define CONFIG_92D_AP_MODE 1
--- a/drivers/staging/rtl8192du/include/hal_intf.h
+++ b/drivers/staging/rtl8192du/include/hal_intf.h
@@ -254,7 +254,7 @@ enum HARDWARE_TYPE {
 
 #define GET_EEPROM_EFUSE_PRIV(priv)	(&priv->eeprompriv)
 
-#ifdef CONFIG_WOWLAN
+#ifdef CONFIG_WAKE_ON_WLAN
 enum wowlan_subcode {
 	WOWLAN_PATTERN_MATCH = 1,
 	WOWLAN_MAGIC_PACKET  = 2,
@@ -285,7 +285,7 @@ struct wowlan_ioctl_param{
 #define Rx_MagicPkt				BIT(5)
 #define FinishBtFwPatch				BIT(7)
 
-#endif /*  CONFIG_WOWLAN */
+#endif /*  CONFIG_WAKE_ON_WLAN */
 
 void rtw_hal_def_value_init(struct rtw_adapter *padapter);
 void rtw_hal_free_data(struct rtw_adapter *padapter);
--- a/drivers/staging/rtl8192du/include/rtl8192d_cmd.h
+++ b/drivers/staging/rtl8192du/include/rtl8192d_cmd.h
@@ -69,16 +69,16 @@ enum RTL8192D_H2C_CMD {
 	H2C_P2P_PS_OFFLOAD = 8,
 	H2C_MAC_MODE_SEL = 9,
 	H2C_PWRM=15,
-#ifdef CONFIG_WOWLAN
+#ifdef CONFIG_WAKE_ON_WLAN
 	H2C_WO_WLAN_CMD = 20,	/*  Wake on Wlan. */
-#endif /*  CONFIG_WOWLAN */
+#endif /*  CONFIG_WAKE_ON_WLAN */
 	H2C_P2P_PS_CTW_CMD = 24,
 	H2C_PathDiv = 26,                  /* PathDiv--NeilChen--2011.07.15 */
-#ifdef CONFIG_WOWLAN
+#ifdef CONFIG_WAKE_ON_WLAN
 	KEEP_ALIVE_CONTROL_CMD=31, /* keep alive for wake on wlan */
 	DISCONNECT_DECISION_CTRL_CMD=32,
 	REMOTE_WAKE_CTRL_CMD=34,
-#endif /*  CONFIG_WOWLAN */
+#endif /*  CONFIG_WAKE_ON_WLAN */
 	H2C_92D_TSF_SYNC=36,
 	H2C_92D_RESET_TSF = 43,
 	H2C_CMD_MAX
@@ -107,7 +107,7 @@ void	rtl8192d_set_p2p_ps_offload_cmd(str
 int reset_tsf(struct rtw_adapter * adapter, u8 reset_port);
 #endif	/*  CONFIG_CONCURRENT_MODE */
 
-#ifdef CONFIG_WOWLAN
+#ifdef CONFIG_WAKE_ON_WLAN
 struct set_wowlan_parm {
 	u8	mode;
 	u8	gpio_index;
@@ -130,6 +130,6 @@ struct set_wowlan_parm {
 
 void rtl8192d_set_wowlan_cmd(struct rtw_adapter* padapter);
 void SetFwRelatedForWoWLAN8192DU(struct rtw_adapter*	padapter,u8 bHostIsGoingtoSleep);
-#endif /*  CONFIG_WOWLAN */
+#endif /*  CONFIG_WAKE_ON_WLAN */
 
 #endif	/*  __RTL8192D_CMD_H_ */
--- a/drivers/staging/rtl8192du/include/rtl8192d_hal.h
+++ b/drivers/staging/rtl8192du/include/rtl8192d_hal.h
@@ -229,10 +229,10 @@ static u32 TargetChnl_2G[TARGET_CHNL_NUM
 struct rt_firmware_92d {
 	u8 *buffer;
 	u32 length;
-#ifdef CONFIG_WOWLAN
+#ifdef CONFIG_WAKE_ON_WLAN
 	u8 *wowbuffer;
 	u32 wowlength;
-#endif /* CONFIG_WOWLAN */
+#endif /* CONFIG_WAKE_ON_WLAN */
 };
 
 /*  This structure must be careful with le byte-ordering */
--- a/drivers/staging/rtl8192du/include/rtw_pwrctrl.h
+++ b/drivers/staging/rtl8192du/include/rtw_pwrctrl.h
@@ -209,14 +209,14 @@ struct pwrctrl_priv
 	u8		bInternalAutoSuspend;
 	u8		bInSuspend;
 	u8		bSupportRemoteWakeup;
-#ifdef CONFIG_WOWLAN
+#ifdef CONFIG_WAKE_ON_WLAN
 	u8		wowlan_mode;
 	u8		wowlan_pattern;
 	u8		wowlan_magic;
 	u8		wowlan_unicast;
 	u8		wowlan_pattern_idx;
 	u32		wowlan_pattern_context[8][5];
-#endif /*  CONFIG_WOWLAN */
+#endif /*  CONFIG_WAKE_ON_WLAN */
 	struct timer_list pwr_state_check_timer;
 	int		pwr_state_check_interval;
 	u8		pwr_state_check_cnts;
--- a/drivers/staging/rtl8192du/os_dep/ioctl_linux.c
+++ b/drivers/staging/rtl8192du/os_dep/ioctl_linux.c
@@ -6888,7 +6888,7 @@ static int rtw_pm_set(struct net_device
 	return ret;
 }
 
-#ifdef CONFIG_WOWLAN
+#ifdef CONFIG_WAKE_ON_WLAN
 static int rtw_wowlan_ctrl(struct net_device *dev,
 			   struct iw_request_info *info,
 			   union iwreq_data *wrqu, char *extra)
@@ -6942,7 +6942,7 @@ _rtw_wowlan_ctrl_exit:
 
 	return ret;
 }
-#endif /* CONFIG_WOWLAN */
+#endif /* CONFIG_WAKE_ON_WLAN */
 
 static int rtw_test(struct net_device *dev, struct iw_request_info *info,
 		    union iwreq_data *wrqu, char *extra)
@@ -7095,11 +7095,11 @@ static const struct iw_priv_args rtw_pri
 		IW_PRIV_TYPE_INT | IW_PRIV_SIZE_FIXED | 2, IW_PRIV_TYPE_CHAR |
 		IW_PRIV_SIZE_FIXED | IFNAMSIZ, "rfr"
 	},
-#ifdef CONFIG_WOWLAN
+#ifdef CONFIG_WAKE_ON_WLAN
 	{
 		SIOCIWFIRSTPRIV + 0xE, 0, 0, "wowlan_ctrl"
 	},
-#endif /*  CONFIG_WOWLAN */
+#endif /*  CONFIG_WAKE_ON_WLAN */
 	{
 		SIOCIWFIRSTPRIV + 0x10,
 		IW_PRIV_TYPE_CHAR | P2P_PRIVATE_IOCTL_SET_LEN, 0, "p2p_set"
@@ -7167,11 +7167,11 @@ static iw_handler rtw_private_handler[]
 	rtw_wx_write_rf,		/* 0x0C */
 	rtw_wx_read_rf,			/* 0x0D */
 
-#ifdef CONFIG_WOWLAN
+#ifdef CONFIG_WAKE_ON_WLAN
 	rtw_wowlan_ctrl,		/* 0x0E */
 #else
 	rtw_wx_priv_null,		/* 0x0E */
-#endif /* CONFIG_WOWLAN */
+#endif /* CONFIG_WAKE_ON_WLAN */
 	rtw_wx_priv_null,		/* 0x0F */
 
 	rtw_p2p_set,			/* 0x10 */
--- a/drivers/staging/rtl8192du/os_dep/usb_intf.c
+++ b/drivers/staging/rtl8192du/os_dep/usb_intf.c
@@ -452,12 +452,12 @@ static void rtw_dev_unload(struct rtw_ad
 
 		/* s5. */
 		if (padapter->bSurpriseRemoved == false) {
-#ifdef CONFIG_WOWLAN
+#ifdef CONFIG_WAKE_ON_WLAN
 			if ((padapter->pwrctrlpriv.bSupportRemoteWakeup==true)&&(padapter->pwrctrlpriv.wowlan_mode==true)) {
 				DBG_8192D("%s bSupportWakeOnWlan==true  do not run rtw_hal_deinit()\n",__func__);
 			}
 			else
-#endif /* CONFIG_WOWLAN */
+#endif /* CONFIG_WAKE_ON_WLAN */
 			{
 				rtw_hal_deinit(padapter);
 			}
@@ -465,9 +465,9 @@ static void rtw_dev_unload(struct rtw_ad
 		}
 
 		padapter->bup = false;
-#ifdef CONFIG_WOWLAN
+#ifdef CONFIG_WAKE_ON_WLAN
 		padapter->hw_init_completed=false;
-#endif /* CONFIG_WOWLAN */
+#endif /* CONFIG_WAKE_ON_WLAN */
 	}
 	else
 	{
@@ -509,9 +509,9 @@ static int rtw_suspend(struct usb_interf
 	struct mlme_priv *pmlmepriv = &padapter->mlmepriv;
 	struct pwrctrl_priv *pwrpriv = &padapter->pwrctrlpriv;
 	struct usb_device *usb_dev = interface_to_usbdev(pusb_intf);
-#ifdef CONFIG_WOWLAN
+#ifdef CONFIG_WAKE_ON_WLAN
 	struct wowlan_ioctl_param poidparam;
-#endif /*  CONFIG_WOWLAN */
+#endif /*  CONFIG_WAKE_ON_WLAN */
 	int ret = 0;
 	u32 start_time = rtw_get_current_time();
 
@@ -535,7 +535,7 @@ static int rtw_suspend(struct usb_interf
 		netif_carrier_off(pnetdev);
 		rtw_netif_stop_queue(pnetdev);
 	}
-#ifdef CONFIG_WOWLAN
+#ifdef CONFIG_WAKE_ON_WLAN
 	if (padapter->pwrctrlpriv.bSupportRemoteWakeup==true&&padapter->pwrctrlpriv.wowlan_mode==true) {
 		u8 ps_mode=PS_MODE_MIN;
 		/* set H2C command */
@@ -543,7 +543,7 @@ static int rtw_suspend(struct usb_interf
 		rtw_hal_set_hwreg(padapter,HW_VAR_WOWLAN,(u8 *)&poidparam);
 	}
 	else
-#endif /* CONFIG_WOWLAN */
+#endif /* CONFIG_WAKE_ON_WLAN */
 	{
 		/* s2. */
 		rtw_disassoc_cmd(padapter, 0, false);
@@ -793,9 +793,9 @@ static void rtw_usb_if1_deinit(struct rt
 	}
 
 	rtw_cancel_all_timer(if1);
-#ifdef CONFIG_WOWLAN
+#ifdef CONFIG_WAKE_ON_WLAN
 	if1->pwrctrlpriv.wowlan_mode=false;
-#endif /* CONFIG_WOWLAN */
+#endif /* CONFIG_WAKE_ON_WLAN */
 	rtw_dev_unload(if1);
 
 	DBG_8192D("%s, hw_init_completed=%d\n", __func__, if1->hw_init_completed);
